test/models/bookmark_test.rb

```diff
@@ -1,6 +1,16 @@
 require 'test_helper'
 
 class BookmarkTest < ActiveSupport::TestCase
+  @valid_controllers ||= (["dashboard", "common_parameters"] +
+      ActiveRecord::Base.connection.tables.map(&:to_s) +
+      Permission.resources.map(&:tableize)).uniq
+
+  should allow_values(*@valid_controllers).for(:controller)
+  should allow_values(*valid_name_list).for(:name)
+  should allow_values(*valid_name_list).for(:query)
+  should_not allow_values(*invalid_name_list).for(:name)
+  should_not allow_values('', ' ').for(:query)
```

Comment:
I think the lines 4-12 need to be in a `test "" do` block or else they may be outside a transaction and cause issues. This also will match the style of every other test.